lib/tests/misc.nix

```
@@ -296,6 +297,30 @@ runTests {
     expected = "a\nb\nc\n";
   };
 
+  testMakeIncludePathWithPkgs = {
+    expr = (makeIncludePath [
+      { dev = { outPath = "/nix/store/bash"; }; }
+      # lib.getOutput "dev" return out if "dev" is not found
+      { out = { outPath = "/nix/store/openssl"; }; }
+    ]);
+    expected = "/nix/store/bash/include:/nix/store/openssl/include";
```

```suggestion
    expr = (makeIncludePath [
      # makeIncludePath preferably selects the "dev" output
      { dev.outPath = "/dev"; out.outPath = "/out"; outPath = "/default"; }
      # "out" is used if "dev" is not found
      { out.outPath = "/out"; outPath = "/default"; }
      # And it returns the derivation directly if there's no "out" either
      { outPath = "/default"; }
      # Same if the output is specified explicitly, even if there's a "dev"
      { dev.outPath = "/dev"; outPath = "/default"; outputSpecified = true; }
    ]);
    expected = "/dev/include:/out/include:/default/include:/default/include";
```

Ideally this would be made into a `getOutput` test, there's no need to test each of the `makeBinPath`, etc. like this individually. It's okay for now though, I take any testing over no testing :)